### ISSUE TITLE
HTTP 500: target not set correct

### ISSUE THREAD

**Issue by discostur** [prometheus-pve/prometheus-pve-exporter#82]
Hi,

just deployed pve_exporter via docker on the proxmox host directly. Exporter works when i try scraping metrics manually via curl: 

```bash
$ curl http://192.168.1.181:9221/pve?target=192.168.1.181
# HELP pve_up Node/VM/CT-Status is online/running
# TYPE pve_up gauge
...
```

However, i configured prometheus like explained but it only gets HTTP 500:

```bash
  - job_name: "test-pve_exporter"
    static_configs:
      - targets: 
        - '192.168.1.181:9221' 
    metrics_path: /pve
    params:
      module: [default]
```

Looks to me like when i curl manually and don't set any target:

```bash
$ curl http://192.168.1.181:9221/pve
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>500 Internal Server Error</title>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>
```

Shouldn't the target been set automatically by prometheus? 

Thanks
Regards,
Kilian

**Comment by discostur** [comment]
Just tried to set the target manually in prometheus config and than it works:

```bash
  - job_name: "test-pve_exporter"
    static_configs:
      - targets: 
        - '192.168.1.181:9221' 
    metrics_path: /pve
    params:
      module: [default]
      target: [192.168.1.181]
```

But i think thats not how it should work ... if i have more than one node its broken again.

**Comment by znerol** [comment]
**TL;DR:** `targets` list has nothing to do with `target` url parameter. This is quite confusing but in prometheus the `targets` list is the source of the `__address__` label.

With this config:

```
  - job_name: "test-pve_exporter"
    static_configs:
      - targets: 
        - '192.168.1.181:9221' 
    metrics_path: /pve
    params:
      module: [default]
```

Prometheus will construct the following endpoint url:

```
http://192.168.1.181:9221/pve
```

You can verify that by navigating to the prometheus web interface and click on the `targets` tab. If you like to propagate the `__address__` (from the `targets` list) to the `target` param, you need `relabel_configs` as shown at the bottom of the readme.

**Comment by discostur** [comment]
@znerol thanks for the explanation. I just tried with different `relabel_configs` but i still can't get it working ... 

My `pve_exporter` is running locally on the proxmox node and `prometheus` is running on a different node: 

prometheus (192.168.1.180) -> proxmox (pve_exporter, 192.168.1.181)

I tried all four configs:

```bash
      - source_labels: [__address__]
        target_label: __param_target
      - source_labels: [__param_target]
        target_label: instance
      - target_label: __address__
        replacement: 127.0.0.1:9221  
```

```bash
      - source_labels: [__address__]
        target_label: __param_target
      - source_labels: [__param_target]
        target_label: instance
      - target_label: __address__
        replacement: 192.168.1.181:9221  
```

```bash
      - source_labels: [__address__]
        target_label: __param_target
      - source_labels: [__param_target]
        target_label: instance
```

```bash
      - source_labels: [__address__]
        target_label: __param_target
```

Can you please point me to the correct `relabel_config`? :/

**Comment by znerol** [comment]
Let's get there systematically. 

> just deployed pve_exporter via docker on the proxmox host directly.

Please check whether PVE API is reachable from within a docker container. Run this from your docker host:

```
docker run -it --rm docker.io/curlimages/curl --include --insecure "https://192.168.1.181:8006/api2/json/access/ticket"
```
This should return something like this:
```
HTTP/1.1 200 OK
Cache-Control: max-age=0
Connection: Keep-Alive
Date: Mon, 30 Aug 2021 17:53:03 GMT
Pragma: no-cache
Server: pve-api-daemon/3.0
Content-Length: 13
Content-Type: application/json;charset=UTF-8
Expires: Mon, 30 Aug 2021 17:53:03 GMT

{"data":null}
```

Also check whether the `--insecure` flag is required. If `curl` succeeds with the `--insecure` flag and if it fails without, then make sure that you need to include `verify_ssl: false` in `pve.yml` configuration file.

Do you manage to get `curl` call working when run from `docker`?

**Comment by discostur** [comment]
Deployed this in another way and it's up and running now - thanks for your help ;)
